src/Nancy.Swagger.Annotations/SwaggerObjects/AnnotatedOperation.cs

```diff
@@ -44,33 +44,72 @@ public AnnotatedOperation(string name, MethodInfo handler, ISwaggerModelCatalog
                 .Select(CreateSwaggerResponseObject)
                 .ToDictionary(x => x.GetStatusCode().ToString(), y => (global::Swagger.ObjectModel.Response) y);
 
-            var infos = handler.GetParameters().Where(x => x.GetCustomAttributes<RouteParamAttribute>().Any()).ToList();
+            var paramsList = new List<Parameter>();
+            //first check if any RouteParamAttributes are on the Method.
+            CreateSwaggerParametersFromMethodAttributes(handler, paramsList);
+            //then check for RouteParamAttributes on themethod parameters
+            CreateSwaggerParametersFromParameters(handler, paramsList);
 
-            Parameters = CreateSwaggerParameters(infos);
+            Parameters = paramsList;
 
         }
 
-        private IEnumerable<Parameter> CreateSwaggerParameters(IEnumerable<ParameterInfo> infos)
+        private void CreateSwaggerParametersFromMethodAttributes(MethodInfo info, List<Parameter> paramsList)
         {
-            var result = new List<Parameter>();
-            foreach (var info in infos)
+            var routeParamsAttributes = info.GetCustomAttributes<RouteParamAttribute>(true);
+            foreach (var attrib in routeParamsAttributes)
             {
-                var paramAttrs = info.GetCustomAttributes<RouteParamAttribute>();
+                if (string.IsNullOrEmpty(attrib.Name))
+                    throw new ArgumentNullException("Name", "RouteParam name cannot be null when used on method.");
+                //we can only have 1 body parameter
+                if (attrib.ParamIn == ParameterIn.Body)
+                {
+                    //if we already have a body parameter then ignore any subsequent ones.
+                    if (paramsList.Where(x => x.GetType() == typeof(AnnotatedBodyParameter)).Any())
+                    {
+                        continue;
+                    }
+                    if ((attrib.ParamType == null))
+                        throw new ArgumentNullException("ParamType", "ParamType for Body must bespecified when RouteParam used on method.");
+
+                    paramsList.Add(new AnnotatedBodyParameter(attrib.Name, attrib.ParamType, attrib, _modelCatalog));
+                }
+                //ignore duplicate named parameters.
+                if (!paramsList.Any(x => x.Name.Equals(attrib.Name)))
+                {
+                    paramsList.Add(new AnnotatedParameter(attrib.Name, attrib.ParamType, attrib));
+                }
+            }
+        }
+
+        private void CreateSwaggerParametersFromParameters(MethodInfo info, List<Parameter> paramsList)
```

Comment:
`CreateSwaggerParametersFromParameterAttributes`? 
It's from both, but it only does it if there are parameter attributes.

Reply:
Not sure what a better name would be? CreateSwaggerParametersFromMethodParametersOnlyIfTheyHaveAttributes?